### PR TITLE
Task should fail properly when required parameters are not found

### PR DESCRIPTION
**Pull request by hiyadav**
#12029

### CODE REVIEW

**Comment by bishal-pdMSFT** [comment]
@jalajmsft please review this

File: Tasks/AzureCLIV2/package-lock.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vsts-tasks-azurecli",
-  "version": "1.0.0",
```
Comment (bishal-pdMSFT) [inline Tasks/AzureCLIV2/package-lock.json]: this version is not needed to be changed. You should change version in task.json

File: Tasks/AzureCLIV2/src/Utility.ts [hunk 2]
```diff
@@ -39,16 +39,16 @@ export class Utility {
         let contents: string[] = [];
         contents.push(`$ErrorActionPreference = '${powerShellErrorActionPreference}'`);
         let filePath: string = tl.getPathInput("scriptPath", false, true);;
-        if (scriptLocation === "inlineScript") {
-            let inlineScript: string = tl.getInput("inlineScript", true);
-            filePath = path.join(tempDirectory, `azureclitaskscript${new Date().getTime()}_inlinescript.${fileExtensions[0]}`);
-            await Utility.createFile(filePath, inlineScript);
-        }
-        else{
+        if (scriptLocation.toLowerCase() === "scriptpath") {
```
Comment (bishal-pdMSFT) [inline Tasks/AzureCLIV2/src/Utility.ts]: What is the scenario this change is trying to fix? scriptLocation is a required input with values either `scriptPath` or `inlineScript`. And for that the existing code should have been fine.

Comment (jalajmsft) [reply Tasks/AzureCLIV2/src/Utility.ts]: +1, what scenario are we trying to fix here?

File: Tasks/AzureCLIV2/azureclitask.ts [hunk 3]
```diff
@@ -46,7 +46,10 @@ export class azureclitask {
             }
         }
         finally {
-            await scriptType.cleanUp();
+            if (scriptType) {
```
Comment (jalajmsft) [inline Tasks/AzureCLIV2/azureclitask.ts]: in yaml case, if scriptType is not set, should we throw an error and fail the task?

File: Tasks/AzureCLIV2/src/ScriptType.ts [hunk 4]
```diff
@@ -7,14 +7,14 @@ export class ScriptTypeFactory {
         let scriptLocation: string = tl.getInput("scriptLocation", true);
```
Comment (jalajmsft) [inline Tasks/AzureCLIV2/src/ScriptType.ts]: in yaml case, should we check scriptLocation is non-empty and value is ["scriptpath", "inlineScript"] else stop execution

File: Tasks/AzureCLIV2/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vsts-tasks-azurecli",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (jalajmsft) [inline Tasks/AzureCLIV2/package.json]: What does this do?

File: Tasks/AzureCLIV2/azureclitask.ts (hunk 3 above)
Comment (hiyadav) [reply Tasks/AzureCLIV2/azureclitask.ts]: Yes, that is the behaviour even today. But it was being overriden by the reference error being generated from the finally block. 

See this line in ScriptType.ts: 
_let scriptType: string = tl.getInput("scriptType", true);_

File: Tasks/AzureCLIV2/package.json (hunk 5 above)
Comment (hiyadav) [reply Tasks/AzureCLIV2/package.json]: reverting this as package.json has not real impact on the task.

File: Tasks/AzureCLIV2/src/Utility.ts (hunk 2 above)
Comment (hiyadav) [reply Tasks/AzureCLIV2/src/Utility.ts]: As the validation for script Location value was missing. If user in YAML pipeline would have entered some wrong input, we would have defaulted to `inlineScript` in getScriptPath function. Thus by changing the order here, i was just ensuring that we always default to `inlineScript`.

But now i have added validation for scriptLocation to always be either of the two allowed values. Hence this change is no more required.

File: Tasks/AzureCLIV2/src/ScriptType.ts (hunk 4 above)
Comment (hiyadav) [reply Tasks/AzureCLIV2/src/ScriptType.ts]: done

**Pull request merged**